pkg/cluster-operator/customcluster_controller_test.go

```diff
@@ -0,0 +1,100 @@
+package clusteroperator
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kurator.dev/kurator/pkg/apis/infra/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestDeleteWorkerPods(t *testing.T) {
+	//create a fake client
+	r := &CustomClusterController{}
+	r.Client = fake.NewClientBuilder().Build()
+	TestcustomCluster := &v1alpha1.CustomCluster{}
+
+	err := r.deleteWorkerPods(context.Background(), TestcustomCluster)
+	if err != nil {
+		t.Errorf("deleteWorkerPods() error is %v", err)
+		return
+	}
+}
+
+func TestWorkerToCustomClusterMapFunc(t *testing.T) {
+	//create a test pod
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "test-pod",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: CustomClusterKind,
+					Name: "test-cluster",
+				},
+			},
+		},
+	}
+
+	r := &CustomClusterController{}
+	r.Client = fake.NewClientBuilder().WithRuntimeObjects(pod).Build()
+
+	// call the function with the pod as input
+	requests := r.WorkerToCustomClusterMapFunc(pod)
+
+	// check that the function returns the expected output
+	expectedRequests := []ctrl.Request{{NamespacedName: client.ObjectKey{Namespace: "default", Name: "test-cluster"}}}
+	assert.Equal(t, expectedRequests, requests)
+}
+
+func TestCustomMachineToCustomClusterMapFunc(t *testing.T) {
+	r := &CustomClusterController{}
+	//create a custom machine
+	machine := &v1alpha1.CustomMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "test",
+			Name:      "test-machine",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "v1",
+					Kind:       CustomClusterKind,
+					Name:       "test",
+					UID:        "",
+				},
+			},
+		},
+	}
+
+	requests := r.CustomMachineToCustomClusterMapFunc(machine)
+
+	assert.Equal(t, 1, len(requests))
+}
+
+func TestClusterToCustomClusterMapFunc(t *testing.T) {
+	Cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "test",
+			Name:      "test-cluster",
+		},
+		Spec: clusterv1.ClusterSpec{
+			InfrastructureRef: &corev1.ObjectReference{
+				Kind:      CustomClusterKind,
+				Namespace: "test",
+				Name:      "test-cluster",
+			},
+		},
+	}
+
+	r := &CustomClusterController{}
+	//r.Client = fake.NewClientBuilder().WithObjects(Cluster).Build()
```

Comment:
The code should be removed instead of being commented out if it is not needed anymore.